backend/plugin/mfa/otp/otp.go

```diff
@@ -0,0 +1,101 @@
+// Package otp is the plugin for OTP Multi-Factor Authentication.
+package otp
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pquerna/otp/totp"
+	"golang.org/x/exp/slices"
+)
+
+const (
+	// issuerName is the name of the issuer of the OTP token.
+	issuerName = "Bytebase"
+	// secretMaxExpiredDuration is the maximum duration that a secret is valid.
+	// The secret is valid for 5 minutes.
+	secretMaxExpiredDuration = 5 * time.Minute
+)
+
+// TimeBasedReader is a reader that returns the same value for the same timestamp.
+type TimeBasedReader struct {
+	reader *strings.Reader
+}
+
+// NewTimeBasedReader creates a new TimeBasedReader for the given timestamp.
+func NewTimeBasedReader(timestamp time.Time) *TimeBasedReader {
+	formatedTimestampUnix := timestamp.Unix() - int64(timestamp.Second())
```

Comment:
What's the rational?

Reply:
In order to generate random strings based on timestamp(minute):
* remove the seconds suffix;
* convert to unix format string;

Reply:
I see. We can do timestamp.Unix() / int(secretMaxExpiredDuration / time.Second), then we only need to loop twice in GetValidSecrets().

Reply:
Updated, PTAL